### PR TITLE
Keep receiving bytes from sockets in socket examples

### PR DESCRIPTION
**Pull request by Acentelles**
`recv` from `Network.Simple.TCP` sometimes receives less bytes than it is supposed to (specially with large payloads), so the client might get a partial encoded message that cannot be deserialized. This is because TCP is stream oriented. We keep receiving bytes from the socket until it is empty

### CODE REVIEW

File: src/Examples/Raft/Socket/Common.hs [hunk 1]
```diff
@@ -9,6 +9,9 @@ import qualified Data.Word8 as W8
 
 import Raft.Types
 
+maxMsgSize :: Int
+maxMsgSize = 1000 * 4096
```
Comment (tdietert) [inline src/Examples/Raft/Socket/Common.hs]: Where does this number come from?

File: src/Examples/Raft/Socket/Common.hs [hunk 2]
```diff
@@ -31,3 +34,14 @@ getFreePort = do
   port <- NS.socketPort sock
   NS.close sock
   pure $ show port
+
+keepReceiving :: N.Socket -> IO (Maybe ByteString)
+keepReceiving sock = do
```
Comment (tdietert) [inline src/Examples/Raft/Socket/Common.hs]: Can we call this `recvAll`? I have seen this function in several libs and think that it follows a good naming convention.

For example: https://hackage.haskell.org/package/socket-0.4.0.1/docs/System-Socket.html#g:16

Comment (tdietert) [reply src/Examples/Raft/Socket/Common.hs]: Can we also pass the message size as an argument to `recvAll`?

File: src/Examples/Raft/Socket/Client.hs [hunk 3]
```diff
@@ -122,7 +122,7 @@ clientResponseServer
 clientResponseServer host port = do
   respChan <- asks clientRespChan
   N.serve (N.Host host) port $ \(sock, _) -> do
-    mBytes <- N.recv sock (4 * 4096)
+    mBytes <- keepReceiving sock
```
Comment (tdietert) [inline src/Examples/Raft/Socket/Client.hs]: Here can we call `keepReceiving` (renamed to `recvAll`) with `maxMsgSize` as an argument?

File: src/Examples/Raft/Socket/Common.hs (hunk 2 above)
Comment (Acentelles) [reply src/Examples/Raft/Socket/Common.hs]: That's a quite old version of that library... It seems that the current `receive` function from `System-Socket` https://hackage.haskell.org/package/socket-0.8.2.0/docs/System-Socket.html#g:12 already keeps retrying until the socket is empty.
Do we want to import a new library for this function?

File: src/Examples/Raft/Socket/Common.hs (hunk 1 above)
Comment (Acentelles) [reply src/Examples/Raft/Socket/Common.hs]: I guess it's just a big number

File: src/Examples/Raft/Socket/Common.hs (hunk 2 above)
Comment (tdietert) [reply src/Examples/Raft/Socket/Common.hs]: sorry, I didn't mean to use `recvAll` from that library, but rather to just rename `keepReceiving` to `recvAll`.

Comment (Acentelles) [reply src/Examples/Raft/Socket/Common.hs]: I see. Yes, sure!

**Comment by tdietert** [comment]
LGTM

**Pull request merged**